### ISSUE TITLE
Errors when removing surfaces from simulation

### ISSUE THREAD

**Issue by knstmrd** [sparta/sparta#414]
## Summary

Several issues appear when removing surfaces from a simulation, exact type of which depends on the read_surf / remove_surf command order.

## Type of Issue

Bug Report

## Detailed Description (Enhancement Suggestion)

Unfortunately I'm not very well-versed in the surface structures in SPARTA, so I have little idea what's causing this behaviour.

## SPARTA Version (Bug Report)

SPARTA cloned from master branch on 31st of May 2023 (last commit 20e2b11).

## Expected Behavior (Bug Report)

Surfaces should be removed without issues (surface persisting or issues popping up later trying to read in the next surface)

## Actual Behavior (Bug Report)

The bug appears when trying to read in 3 surfaces, and later deleting 2 of them. All 3 surfaces are assigned to different groups.
Different errors pop up depending on the order of `read_surf` and `remove_surf` commands:

1. If all 3 surfaces are read in at first, and then surfaces number 2 ("refinementsurfb") and 1 ("refinementsurfa") are deleted (in that order) ("option 3" in the code comments), then the following error appears `ERROR: Watertight check failed with 8 duplicate points (../surf.cpp:1171)`
2. If only the first 2 surfaces are read in (with surface 3 being read in after their deletion), and then surfaces number 2 and 1 are deleted (in that order) (option 3 in the code), then the following error appears when reading in surface 3 ("discsurf"): `ERROR: Read_surf minimum surface ID is 5 (../read_surf.cpp:2054)`
3. If one deletes first surface 1 and then surface 2 ("option 2" in the code comments), then surface 1 is not removed (as can be seen on the image dump), calling `remove_surf refinementsurfa` a second time remedies that. So no errors in this case, but the behaviour is erroneous.
4. Removing all surfaces via `remove_surf all` works as expected.

## Steps to Reproduce (Bug Report)

Input files (with comments) and surface data attached.

## Further Information, Files, and Links

```
seed	    	    12345
dimension   	    2
global              gridcut -1.0 comm/sort no

boundary	    oo ao pp

create_box  	    -1.0 2.0 0.0 2.0 -0.5 0.5
create_grid 30 20 1

surf_collide bcspecular specular

read_surf square_1.surf clip group refinementsurfa
read_surf square_2.surf clip group refinementsurfb

surf_modify refinementsurfa collide bcspecular 
surf_modify refinementsurfb collide bcspecular 

# if placed here and used with option 3, watertight check fails
# read_surf square_3.surf clip group discsurf
# surf_modify discsurf collide bcspecular

# option 1: works OK
# remove_surf all

# option 2: does not remove refinementsurfa and needs a second remove_surf call
# remove_surf refinementsurfa
# remove_surf refinementsurfb
# remove_surf refinementsurfa

# option 3: breaks code
remove_surf refinementsurfb
remove_surf refinementsurfa

read_surf square_3.surf clip group discsurf
surf_modify discsurf collide bcspecular

# species		    ar.species Ar
# mixture		    air Ar vstream 0.0 0.0 0.0 temp 273.15

global              nrho 7.07043E22
global              fnum 7.07043E6

balance_grid        rcb part

stats		    100

stats_style	    step cpu


dump                2 image all 50 grid_and_surf_refined*.ppm type type pdiam 0.002 &
	            size 2048 2048 particle yes &
                box yes 0.01 &
                    gline yes 0.005 &
                    surf proc 0.01 zoom 0.9
dump_modify	    2 pad 4 


timestep 	    7.00E-9
run 		    1
```

Input files + surface data
[example.tar.gz](https://github.com/sparta/sparta/files/11611630/example.tar.gz)

**Comment by stanmoore1** [comment]
@knstmrd thanks for the report, will take a look.

**Comment by stanmoore1** [comment]
I can reproduce the issue

**Comment by stanmoore1** [comment]
@knstmrd I believe I fixed this issue with #415, can you confirm?
